src/main/java/org/openrewrite/java/spring/boot3/MigrateWebMvcTagsToObservationConvention.java

```diff
@@ -0,0 +1,285 @@
+                List<Statement> getTagsBodyStatements = new ArrayList<>();
+                for (Statement stmt : classDecl.getBody().getStatements()) {
+                    if (stmt instanceof J.MethodDeclaration) {
+                        J.MethodDeclaration md = (J.MethodDeclaration) stmt;
+                        if (md.getSimpleName().equals("getTags") && md.getBody() != null) {
+                            getTagsBodyStatements.addAll(md.getBody().getStatements().subList(0, md.getBody().getStatements().size() - 1));
+                            Statement ret = md.getBody().getStatements().get(md.getBody().getStatements().size() - 1);
+                            if (ret instanceof J.Return && ((J.Return) ret).getExpression() instanceof J.MethodInvocation) {
+                                if (TAGS_OF_ANY.matches(((J.Return) ret).getExpression())) {
+                                    getTagsBodyStatements.add((Statement) ((J.Return) ret).getExpression());
+                                }
+                            }
+                            break;
+                        }
+                    }
+                }
+
+                String tmpl = "class " + classDecl.getSimpleName() + " extends DefaultServerRequestObservationConvention {\n" +
+                              "    @Override\n" +
+                              "    public KeyValues getLowCardinalityKeyValues(ServerRequestObservationContext context) {\n" +
+                              "        KeyValues values = super.getLowCardinalityKeyValues(context);\n" +
+                              "        return values;" +
+                              "    }\n" +
+                              "}";
+                J.ClassDeclaration newClassDeclaration = JavaTemplate.builder(tmpl)
+                        .contextSensitive()
+                        .javaParser(JavaParser.fromJavaVersion().classpathFromResources(ctx, "micrometer-commons", "spring-web-6.+", "jakarta.servlet-api"))
+                        .imports(DEFAULTSERVERREQUESTOBSERVATIONCONVENTION_FQ, KEYVALUES_FQ, HTTPSERVLETREQUEST_FQ, HTTPSERVLETRESPONSE_FQ, SERVERREQUESTOBSERVATIONCONVENTION_FQ)
+                        .build()
+                        .apply(getCursor(), classDecl.getCoordinates().replace());
+
+                J.ClassDeclaration finalNewClassDeclaration = newClassDeclaration;
+                newClassDeclaration = newClassDeclaration
+                        .withLeadingAnnotations(classDecl.getLeadingAnnotations())
+                        .withModifiers(classDecl.getModifiers())
+                        .withPrefix(classDecl.getPrefix())
+                        .withBody(newClassDeclaration.getBody().withStatements(ListUtils.map(classDecl.getBody().getStatements(), stmt -> {
+                            if (stmt instanceof J.MethodDeclaration && ((J.MethodDeclaration) stmt).getSimpleName().equals("getTags")) {
```

Comment:
Here we could check if it is referentially equal to that `md` we previously found.